app/src/main/java/de/s3xy/architecturesample/login/ui/LoginActivity.java

```diff
@@ -0,0 +1,119 @@
+package de.s3xy.architecturesample.login.ui;
+
+import android.annotation.TargetApi;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.view.View;
+import android.webkit.WebResourceRequest;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+
+import javax.inject.Inject;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.Unbinder;
+import de.s3xy.architecturesample.AwesomeApplication;
+import de.s3xy.architecturesample.R;
+import de.s3xy.architecturesample.base.BaseActivity;
+import de.s3xy.architecturesample.github.GithubAuthHelper;
+import de.s3xy.architecturesample.login.presenter.LoginPresenter;
+import de.s3xy.architecturesample.search.ui.SearchRepositoriesActivity;
+
+/**
+ * Created by Vlad Fedorenko <vfedo92@gmail.com>
+ * 08.25.2016
+ */
+public class LoginActivity extends BaseActivity implements LoginView {
+
+    @Inject
+    LoginPresenter mLoginPresenter;
+
+    @BindView(R.id.web_view)
+    WebView mWebView;
+
+    private Unbinder mUnbinder;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        ((AwesomeApplication) getApplication()).getApplicationComponent().inject(this);
+
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+
+        mUnbinder = ButterKnife.bind(this);
+        mLoginPresenter.attachView(this);
+    }
+
+    protected Unbinder getUnbinder() {
+        return mUnbinder;
+    }
+
+    protected LoginPresenter getPresenter() {
+        return mLoginPresenter;
+    }
+
+    public void onSignInClick(View view) {
+        mLoginPresenter.signIn();
+    }
+
+    public void onSkipClick(View view) {
+        mLoginPresenter.skipLogin();
+    }
+
+    @Override
+    public void showWebViewForLogin(String url) {
```

Comment:
The link you provided describes only authorization flow, not authentication. But GitHub gives us opportunity to get access token with Basic Authorization when we enter login and pass within the app. But it brakes OAuth protocol rule (Wiki says: "Generally, OAuth provides to clients a "secure delegated access" to server resources on behalf of a resource owner. It specifies a process for resource owners to authorize third-party access to their server resources without sharing their credentials")
If I got it right, our app must not know about any credentials, user should enter it on GitHub side. So I strongly believe we need WebView in any OAuth application.